crates/wasm-mutate/src/mutators/peephole/eggsy/expr_enumerator.rs

```diff
@@ -0,0 +1,877 @@
+use std::cell::RefCell;
+
+use crate::mutators::peephole::{
+    eggsy::{encoder::rebuild::build_expr_inner, RandomExtractor},
+    EG,
+};
+use egg::{AstSize, Id, RecExpr};
+use rand::{prelude::SmallRng, Rng};
+
+use super::lang::Lang;
+
+macro_rules! binop {
+    ($lang:ident, $left:ident, $right: ident, $egraph: ident, $rnd: ident, $depth: ident, $recexpr: ident) => {{
```

Comment:
Is there a reason that these need to be macros and can't be functions?

Also, some doc comments would be really appreciated here.

Reply:
Mostly to avoid repetition

Reply:
Which repetition can't be avoided with a helper function rather than a helper macro? AFAICT, a function should work here.

Reply:
Well, not only that, I a using them to generate the right no construction `Lang::$lang`